build/media_source/com_joomlaupdate/js/admin-update-default.es6.js

```diff
@@ -3,13 +3,226 @@
+    const messageDiv = document.getElementById('errorDialogMessage');
+    const progressDiv = document.getElementById('progress-bar');
+    const titleDiv = document.getElementById('update-title');
+    const helpDiv = document.getElementById('update-help');
+
+    progressDiv.classList.add('bg-danger');
+    progressDiv.classList.remove('bg-success');
+    titleDiv.innerHTML = Joomla.Text._('COM_JOOMLAUPDATE_UPDATING_FAIL');
+    helpDiv.classList.remove('d-none');
+    helpDiv.classList.add('d-grid');
+
+    if (isForbidden) {
+      headerDiv.innerHTML = Joomla.Text._('COM_JOOMLAUPDATE_ERRORMODAL_HEAD_FORBIDDEN');
+      messageDiv.innerHTML = Joomla.Text._('COM_JOOMLAUPDATE_ERRORMODAL_BODY_FORBIDDEN');
+    } else {
+      headerDiv.innerHTML = Joomla.Text._('COM_JOOMLAUPDATE_ERRORMODAL_HEAD_SERVERERROR');
+      messageDiv.innerHTML = Joomla.Text._('COM_JOOMLAUPDATE_ERRORMODAL_BODY_SERVERERROR');
+    }
+
+    const myModal = new bootstrap.Modal(document.getElementById('errorDialog'), {
+      keyboard: true,
+      backdrop: true,
+    });
+    myModal.show();
+  },
+  startExtract: () => {
+    // Reset variables
+    Joomla.Update.stat_files = 0;
+    Joomla.Update.stat_inbytes = 0;
+    Joomla.Update.stat_outbytes = 0;
+
+    const postData = new FormData();
+    postData.append('task', 'startExtract');
+    postData.append('password', Joomla.Update.password);
+
+    // Make the initial request to the extraction script
+    Joomla.request({
+      url: Joomla.Update.ajax_url,
+      data: postData,
+      method: 'POST',
+      perform: true,
+      onSuccess: (rawJson) => {
+        try {
+          // If we can decode the response as JSON step through the update
+          const data = JSON.parse(rawJson);
+          Joomla.Update.stepExtract(data);
+        } catch (e) {
+          // Decoding failed; display an error
+          Joomla.Update.genericErrorMessage(e.message);
+        }
+      },
+      onError: Joomla.Update.handleErrorResponse,
+    });
+  },
+  stepExtract: (data) => {
+    // Did we get an error from the ZIP extraction engine?
+    if (data.status === false) {
+      Joomla.Update.genericErrorMessage(data.message);
+
+      return;
+    }
+
+    const progressDiv = document.getElementById('progress-bar');
+    const titleDiv = document.getElementById('update-title');
+
+    // Are we done extracting?
+    if (data.done) {
+      progressDiv.classList.add('bg-success');
+      progressDiv.style.width = '100%';
+      progressDiv.setAttribute('aria-valuenow', 100);
+      titleDiv.innerHTML = Joomla.Text._('COM_JOOMLAUPDATE_UPDATING_COMPLETE');
+
+      Joomla.Update.finalizeUpdate();
+
+      return;
+    }
+
+    // Add data to variables
+    Joomla.Update.stat_inbytes = Joomla.Update.formatBytes(data.bytesIn);
+    Joomla.Update.stat_percent = data.percent;
+    Joomla.Update.stat_percent = Joomla.Update.stat_percent
+      || (100 * (Joomla.Update.stat_inbytes / Joomla.Update.totalsize));
```

Comment:
```suggestion
    Joomla.Update.stat_percent = data.percent;
    Joomla.Update.stat_percent = Joomla.Update.stat_percent
      || (100 * (Joomla.Update.stat_inbytes / Joomla.Update.totalsize));
    Joomla.Update.stat_inbytes = Joomla.Update.formatBytes(data.bytesIn);
```

@nikosdion Here you assign the formatted bytes, i.e. including the engineering unit, to `Joomla.Update.stat_inbytes`, and use that later to calculate `Joomla.Update.stat_percent` in case if that has no value yet. I#m not sure if that works.

So maybe you should change the order here as suggested.

But maybe it would be better to use the numeric, unformatted values in `Joomla.Update.stat_inbytes` and `Joomla.Update.stat_outbytes` and do the formatting when assigning it to the `innerText` of the `span` elements when updating the display?

Reply:
But it also can be that I'm wrong here and just don't understand the code. All I can say in my tests I did never see any bytes.

Reply:
@nikosdion I have made a PR in your repo for fixing this here and the problem with the counters not being shown: https://github.com/nikosdion/joomla-cms/pull/1

Reply:
I resolve this conversation because the suggestion is not good. My PR does it better.